stripe/api_resources/abstract/test_helpers.py

```diff
@@ -0,0 +1,49 @@
+from __future__ import absolute_import, division, print_function
+
+from stripe import error, util, six
+from stripe.six.moves.urllib.parse import quote_plus
+
+
+class APIResourceTestHelpers:
+    def __init__(self, resource):
+        self.resource = resource
+
+    @classmethod
+    def class_url(cls):
+        if cls == APIResourceTestHelpers:
+            raise NotImplementedError(
+                "APIResourceTestHelpers is an abstract class.  You should perform "
+                "actions on its subclasses (e.g. Charge, Customer)"
+            )
+        # Namespaces are separated in object names with periods (.) and in URLs
+        # with forward slashes (/), so replace the former with the latter.
+        base = cls._resource_cls.OBJECT_NAME.replace(".", "/")
+        return "/v1/test_helpers/%ss" % (base,)
+
+    def instance_url(self):
+        id = self.resource.get("id")
+
+        if not isinstance(id, six.string_types):
+            raise error.InvalidRequestError(
+                "Could not determine which URL to request: %s instance "
+                "has invalid ID: %r, %s. ID should be of type `str` (or"
+                " `unicode`)" % (type(self).__name__, id, type(id)),
+                "id",
+            )
+
+        id = util.utf8(id)
+        base = self.class_url()
+        extn = quote_plus(id)
+        return "%s/%s" % (base, extn)
+
+
+def test_helpers(cls):
+    def test_helpers_getter(self):
+        return self.TestHelpers(self)
+
+    cls.TestHelpers._resource_cls = cls
+    cls.TestHelpers._static_request = cls._static_request
+    cls.TestHelpers._static_request_stream = cls._static_request_stream
```

Comment:
I wonder if we should share some base class/mixin between this and `APIResource` so that we don't have to do this. In the interim, let's throw if cls isn't an instance of `APIResource`?